### PR TITLE
Updates to build scripts and manifests (packaging)

### PR DESCRIPTION
**Pull request by gavanderhoorn**
As per subject.

Most of this was triggered by output of `catkin_lint -W2` or by running into a non-functioning install space.

### CODE REVIEW

**Comment by gavanderhoorn** [comment]
7f1cbf2 fixes an issue with `khi_robot_control` that appears to be preventing the ROS buildfarm from building that package successfully ([job1 log](http://build.ros.org/view/Ksrc_uX/job/Kbin_uX64__khi_robot_control__ubuntu_xenial_amd64__binary/1/console)).

**Comment by gavanderhoorn** [comment]
An unresolved issue is how to deal with `install(..)`-ing the `main` and `khi_robot_client` targets in `khi_robot_control`: because of their reliance on the KRNX binary blob, they could pose a problem.

`libkrnx-*.so` is not installed at the moment, and linking is done against the file in`khi_robot_control/lib`, which is located in the source space of the workspace. Copying the `.so` file to the appropriate location using `install(..)` is possible, but that doesn't update the `RPATH` of `khi_robot_control`, causing a runtime dynamic linker failure.

But without installing `main` and `khi_robot_client`, the `khi_robot_control` package, as released through the ROS buildfarm is useless.

At the moment the driver can only be used from a devel or build space, or a local install space.

**Comment by d-nakamichi** [comment]
Thank you for your contribution about ROS build/install.
I made this driver only for a local install space, and this is a very helpful PR.

About `khi_robot_control`, it must need krnx library (not OSS).
So I have 3 ideas to solve this problem. 

1. Use a krnx library as a static library `.a` in `khi_robot_control`
2. Make a new ROS package like `khi_robot_krnx` including `*.so` and `khi_robot_control` use it by `*_depend`
3. When executing `install` in  `khi_robot_control`, copy `*.so` into `CATKIN_PACKAGE_LIB_DESTINATION`

Do you have any ideas ?

**Comment by gavanderhoorn** [comment]
> About `khi_robot_control`, it must need krnx library (not OSS).
> So I have 3 ideas to solve this problem.
> 
>     1. Use a krnx library as a static library `.a` in `khi_robot_control`

This would certainly work around any issues with dynamic linking.

>     2. Make a new ROS package like `khi_robot_krnx` including `*.so` and `khi_robot_control` use it by `*_depend`

This could also work, and would make depending on the library from other ROS packages easier.

>     3. When executing `install` in  `khi_robot_control`, copy `*.so` into `CATKIN_PACKAGE_LIB_DESTINATION`

This last one could be problematic because of the `RPATH` issue I described above. I'm not an expert though, so perhaps this could be easily fixed. Adding `install(..)` for the `krnx-*.so` files was the first thing I did, but then ran into linking problems.

> Do you have any ideas ?

the static library approach would probably be the easiest way to work-around any linking issues.

If you could submit a PR adding the static libraries to the repository I'll rebase this PR.

**Comment by d-nakamichi** [comment]
OK, I'll submit a PR based on idea 1.
Thank you.

File: khi_robot_control/CMakeLists.txt [hunk 1]
```diff
@@ -20,36 +32,35 @@ include_directories(
...
 endif()
 
+add_library(krnx_lib SHARED IMPORTED)
+set_target_properties(krnx_lib PROPERTIES
+  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/lib/lib${krnx_lib_name}.so)
+set(krnx_LIBRARIES krnx_lib)
+
 ## Declare a C++ library
-add_library(khi_robot_client src/khi_robot_krnx_driver.cpp src/khi_robot_client.cpp )
+add_library(khi_robot_client src/khi_robot_client.cpp src/khi_robot_krnx_driver.cpp)
```
Comment (d-nakamichi) [inline khi_robot_control/CMakeLists.txt]: I executed `catkin_make` in this revision, and I got include error `khi_robot_msgs/KhiRobotCmd.h`.
Maybe it needs `add_dependencies` about srv.

```suggestion
add_dependencies(khi_robot_client ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
```

File: rs080n_moveit_config/CMakeLists.txt [hunk 2]
```diff
@@ -8,3 +8,4 @@ catkin_package()
 install(DIRECTORY launch DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
   PATTERN "setup_assistant.launch" EXCLUDE)
 install(DIRECTORY config DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
+install(DIRECTORY script DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
```
Comment (d-nakamichi) [inline rs080n_moveit_config/CMakeLists.txt]: `rs080n_moveit_config` doesn't have a script directory.
So this is not needed.

File: khi_robot_control/CMakeLists.txt (hunk 1 above)
Comment (gavanderhoorn) [reply khi_robot_control/CMakeLists.txt]: I've dropped the commit that introduced this change.

I only build with `catkin_tools`, and that seems to order targets slightly differently, so I did not run into this.

File: rs080n_moveit_config/CMakeLists.txt (hunk 2 above)
Comment (gavanderhoorn) [reply rs080n_moveit_config/CMakeLists.txt]: Unfortunate copy-pasta (I assumed all moveit configs had the example script).

**Comment by gavanderhoorn** [comment]
@d-nakamichi: just build-tested this with `catkin_make`: apart from the KRNX library issue this now seems to work.

**Comment by gavanderhoorn** [comment]
I would actually suggest to fix the KRNX library linking issue in a later PR.

That way we can verify the comment by @k-okada in https://github.com/gavanderhoorn/khi_robot/pull/1#issue-266879413 and not hold up this PR (which fixes an issue that is currently blocking the ROS buildfarm).

**Comment by d-nakamichi** [comment]
Yes, that's true.
I checked the problems are fixed.

**Comment by gavanderhoorn** [comment]
@d-nakamichi, @k-okada: I've just added 6641def which adds something like what @k-okada suggested in https://github.com/gavanderhoorn/khi_robot/pull/1#issue-266879413.

As the commit message says, I'm not sure this is a proper solution, but it does seem to work for me on the systems that didn't want to run `main` before.

@k-okada: could you verify this also works for you?

**Comment by d-nakamichi** [comment]
I accepted the PR. Thank you! @gavanderhoorn

**Comment by gavanderhoorn** [comment]
@d-nakamichi: is squash-merge the default merging policy for this repository? Provenance of commits and history is lost to a large degree that way.

**Pull request merged**